tests/platform_tests/api/test_chassis.py

```diff
@@ -41,7 +52,50 @@
 
 
 class TestChassisApi(PlatformApiTestBase):
-    ''' Platform API test cases for the Chassis class'''
+    """Platform API test cases for the Chassis class"""
+
+    chassis_truth = None
+    duthost_vars = None
+
+    @pytest.fixture(scope="function", autouse=True)
+    def setup(self, duthost):
+        # Get platform truths from platform.json file
+        self.chassis_truth = duthost.facts.get("chassis")
+        if not self.chassis_truth:
+            logger.warning("Unable to get chassis_truth from platform.json, test results will not be comprehensive")
+
+        # Get host vars from inventory file
+        self.duthost_vars = duthost.host.options['inventory_manager'].get_host(duthost.hostname).vars
+
+    #
+    # Helper functions
+    #
+
+    def compare_value_with_platform_facts(self, key, value):
```

Comment:
Should we keep these comparison functions in every  device test_case file, because similar check would be required everywhere ?  Do we simply put this inside a common definition ?

Reply:
I'm working on this at the moment :) Trying to devise a common method as best as possible.If you have any suggestions, please share.

Reply:
One issue with making this function generic is that this test is checking for keys directly under the "chassis" element of the platform.json data. However, other tests will need to access deeper nested values.

Reply:
I was thinking of two ways : 
1) we could separate the logic of value retrieval from platform.json in one common function and 
    expect could be used as is , we are going to  need to pass duthost to retrieve the actual value in json 
2) we could create a new expect utility which handles everything from retrieving to comparison, 
In both the approaches we also need to pass a param of whether we are trying to retrieve from inventory file or json

Reply:
Hmm, We could pass the truth from the fixture itself, for example chassis_truth for chassis tests , fan_truth could be passed in fan related tests, but then maybe it defeats the purpose , because error checking for truth presence has already happened before retrieving.

Reply:
I'm not sure I understand what you mean by "pass the truth from the fixture itself".

Reply:
I meant we could retrieve the actual dict instance of the device itself, at whichever layer it is nested inside platform.json and then pass that to the expect utility  example fan_truth, chassis_truth, thermal_truth etc.

Reply:
Sorry, but I'm still not fully understanding this. It might be better to take this discussion offline.

I think for now I'm going to isolate the changes to this file, and as we begin adding similar functions to the other classes, if we find a pattern, we can create another PR to clean it up by sharing code.

Reply:
Sure sounds good.